src/main/java/io/kestra/plugin/azure/datafactory/CreateRun.java

```diff
@@ -0,0 +1,203 @@
+                    pipelineName: examplePipeline
+                    resourceGroupName: exampleResourceGroup
+                    subscriptionId: 12345678-1234-1234-1234-12345678abc
+                    tenantId: "{{ secret('DATAFACTORY_TENANT_ID') }}"
+                    clientId: "{{ secret('DATAFACTORY_CLIENT_ID') }}"
+                    clientSecret: "{{ secret('DATAFACTORY_CLIENT_SECRET') }}"
+                """
+                )
+        }
+)
+@Schema(
+        title = "Pipeline CreateRun.",
+        description = "Launch an Azure DataFactory pipeline from Kestra"
+)
+public class CreateRun extends AbstractDataFactoryConnection implements RunnableTask<CreateRun.Output> {
+    private static final String PIPELINE_SUCCEEDED_STATUS = "Succeeded";
+
+    @Schema(title = "Factory name")
+    private Property<String> factoryName;
+
+    @Schema(title = "Pipeline name")
+    private Property<String> pipelineName;
+
+    @Schema(title = "Resource Group name")
+    private Property<String> resourceGroupName;
+
+    @Schema(
+            title = "Pipeline parameters."
+    )
+    @Builder.Default
+    private Property<Map<String, Object>> parameters = Property.of(new HashMap<>());
+
+    @Override
+    public CreateRun.Output run(RunContext runContext) throws Exception {
+        Logger logger = runContext.logger();
+
+        //Authentication
+        DataFactoryManager manager = this.getDataFactoryManager(runContext);
+        logger.info("Successfully authenticate to Azure Data Factory");
+
+        //Create running pipeline
+        final String resourceGroupName = this.resourceGroupName.as(runContext, String.class);
+        final String factoryName = this.factoryName.as(runContext, String.class);
+        final String pipelineName = this.pipelineName.as(runContext, String.class);
+
+        final var pipelineRunResponse = manager.pipelines()
+                .createRunWithResponse(resourceGroupName,
+                        factoryName,
+                        pipelineName,
+                        null,
+                        null,
+                        null,
+                        null,
+                        parameters.asMap(runContext, String.class, Object.class),
+                        Context.NONE
+                );
+
+        if(pipelineRunResponse.getStatusCode() != HttpResponseStatus.OK.code()) {
+            throw new RuntimeException();
+        }
+        logger.info("Created run for pipeline '{}'", pipelineName);
+
+        // Get running pipeline
+        final String runId = pipelineRunResponse.getValue().runId();
+
+        PipelineRun pipelineRun = getRunningPipeline(resourceGroupName,factoryName, runId, manager);
+        String runStatus = pipelineRun.status();
+
+        while ("InProgress".equals(runStatus)) {
+            Thread.sleep(1000);
+            pipelineRun = getRunningPipeline(resourceGroupName,factoryName, runId, manager);;
+            runStatus = pipelineRun.status();
+        }
```

Comment:
This is a little too naive with a risk of being stuck indefinitely.
What we usually do is to have two configuration duration:
- completionCheckInterval: duration to sleep between two calls (usually 500ms but it depends on the minimal latency to execute the pipeline)
- waitUntilCompletion: duration after which the task is considered as failed

Each task that is execution on the worker uses one of the worker thread so we cannot risk to have zombie threads.

You can take as inspiration the following task: https://github.com/kestra-io/plugin-ee-aws/blob/master/src/main/java/io/kestra/plugin/ee/aws/runner/Batch.java

Reply:
Thanks. I knew it was a little bit too naive as you say. 
It looks interesting. I am going to implement it